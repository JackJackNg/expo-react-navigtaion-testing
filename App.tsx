import * as React from 'react';
import { Platform, StatusBar, StyleSheet, View } from 'react-native';
import { AppLoading, Asset, Font, Icon } from 'expo';
import AppNavigator from './navigation/AppNavigator';

interface P {
  skipLoadingScreen?: boolean;
}

interface S {
  isLoadingComplete: boolean;
}

export default class App extends React.Component<P, S> {
  state = {
    isLoadingComplete: false,
  };

  render() {
    if (!this.state.isLoadingComplete && !this.props.skipLoadingScreen) {
      return (
        <AppLoading
          startAsync={this._loadResourcesAsync}
          onError={this._handleLoadingError}
          onFinish={this._handleFinishLoading}
        />
        // <View/>
      );
    }
    return (
      <View style={styles.container}>
        {Platform.OS === 'ios' && <StatusBar barStyle='default' />}
        <AppNavigator />
      </View>
    );
  }

  // tslint:disable-next-line:variable-name
  _loadResourcesAsync = async () => {
    await Promise.all([
      Asset.loadAsync([
        require('./assets/images/robot-dev.png'),
        require('./assets/images/robot-prod.png'),
      ]),
      Font.loadAsync({
        // This is the font that we are using for our tab bar
        ...Icon.Ionicons.font,
        // We include SpaceMono because we use it in HomeScreen.js. Feel free
        // to remove this if you are not using it in your app
        'space-mono': require('./assets/fonts/SpaceMono-Regular.ttf'),
      }),
    ]);
    return;
  }

  // tslint:disable-next-line:variable-name
  _handleLoadingError = (error: Error) => {
    // In this case, you might want to report the error to your error
    // reporting service, for example Sentry
    console.warn(error);
  }

  // tslint:disable-next-line:variable-name
  _handleFinishLoading = () => {
    this.setState({ isLoadingComplete: true });
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});
