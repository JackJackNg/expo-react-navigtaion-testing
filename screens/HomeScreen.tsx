import React from 'react';
import {
  Text,
  View,
  StyleSheet,
} from 'react-native';
import Layout from '../constants/Layout';

export default class HomeScreen extends React.Component {
  static navigationOptions = {
    header: null,
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>Hello, this support typescript :)</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    alignContent: 'center',
    justifyContent: 'center',
    width: Layout.window.width,
    height: Layout.window.height,
  },
  text: {
    textAlign: 'center',
  },
});
