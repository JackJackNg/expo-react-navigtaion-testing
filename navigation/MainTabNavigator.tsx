import { createStackNavigator, createDrawerNavigator } from 'react-navigation';

import HomeScreen from '../screens/HomeScreen';

const HomeStack = createStackNavigator({
  Home: HomeScreen,
});

HomeStack.navigationOptions = {};

export default HomeStack;
