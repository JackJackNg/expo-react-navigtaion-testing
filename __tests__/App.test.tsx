// import * as React from 'react';
// import * as renderer from 'react-test-renderer';
// import App from '../App';
// import HomeScreen from '../screens/HomeScreen';

// describe('App', () => {
//   it('snapshot test', () => {
//     const tree = renderer.create(<App skipLoadingScreen={true} />).toJSON;
//     // const tree = renderer.create(<HomeScreen />).toJSON;
//     expect(tree).toMatchSnapshot();
//   });
// });

import 'react-native';
import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';
// @ts-ignore
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import HomeScreen from '../screens/HomeScreen';
import MainTabNavigator from '../navigation/MainTabNavigator';

describe('App snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders the loading screen', async () => {
    const tree = renderer.create(<MainTabNavigator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('renders the root without loading screen', async () => {
  //   const tree = renderer.create(<App skipLoadingScreen={true} />).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
