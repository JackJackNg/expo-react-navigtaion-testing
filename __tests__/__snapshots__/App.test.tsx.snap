// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders the loading screen 1`] = `null`;

exports[`App snapshot test 1`] = `[Function]`;
